game-headed/src/main/java/org/triplea/game/client/parser/DefaultGameLoader.java

```diff
@@ -0,0 +1,36 @@
+package org.triplea.game.client.parser;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.GameParseException;
+import games.strategy.engine.framework.ui.GameChooserEntry;
+import java.util.function.Consumer;
+import javafx.concurrent.Service;
+import javafx.concurrent.Task;
+
+public class DefaultGameLoader extends Service<GameData> implements GameLoader {
+  private GameChooserEntry gameChooserEntry;
+
+  @Override
+  public void loadGame(
+      final GameChooserEntry gameChooserEntry,
+      final Consumer<GameData> onLoad,
+      final Consumer<Throwable> errorHandler) {
+    this.gameChooserEntry = gameChooserEntry;
+    reset();
+    start();
+    setOnSucceeded(value -> onLoad.accept(getValue()));
+    setOnFailed(value -> errorHandler.accept(getException()));
+  }
+
+  @Override
+  protected Task<GameData> createTask() {
+    return new Task<>() {
+      @Override
+      protected GameData call() throws GameParseException {
+        return gameChooserEntry
+            .getCompleteGameData()
+            .orElseThrow(() -> new GameParseException("Missing file!"));
```

Comment:
I can't remember the exact details but I think the reason I chose `GameParseException` here is so that the exception type is consistent.
I thought that I had a try catch block somewhere where this would make sense, but we just have a generic error handler, so yeah any exception would do.

Reply:
How about changing to `FileNotFound` or a `IllegalState`? `GameParseException` AFAIK is more about invalid syntax rather than missing game file.